rmf_api_msgs/schemas/task_state.schema.json

```diff
@@ -0,0 +1,137 @@
+{
+  "$schema": "https://json-schema.org/draft/2020-12/schema",
+  "$id": "https://open-rmf.org/rmf_api_msgs/task_state/0.1",
+  "title": "Task State",
+  "description": "The state of a task",
+  "type": "object",
+  "properties": {
+    "booking": { "$ref": "#/$defs/booking" },
+    "header": { "$ref": "#/$defs/header" },
+    "completed": {
+      "description": "An array of the completed phases of this task",
+      "type": "array",
+      "items": { "$ref": "#/$defs/completed_phase" }
+    },
+    "active": { "$ref": "#/$defs/phase" },
+    "pending": {
+      "description": "An array of the pending phases of this task",
+      "type": "array",
+      "items": { "$ref": "#/$defs/pending_phase" }
+    }
+  },
+  "$defs": {
+    "completed_phase": {
+      "description": "Information about a phase that has completed",
+      "type": "object",
+      "properties": {
+        "phase": { "$ref": "#/$defs/phase" },
+        "unix_millis_start_time": { "type": "integer" },
+        "unix_millis_finish_time": { "type": "integer" }
+      }
+    },
+    "phase": {
+      "description": "Information about a phase",
+      "type": "object",
+      "properties": {
+        "id": { "$ref": "#/$defs/id" },
```

Comment:
Is `id` unique for a task or across all tasks?

This affects how it can be stored into a database and how it is queried.

Reply:
The phase ID is unique _within_ each task, that way tasks don't need to worry about collisions with other tasks. For storing in a database we'll need to key it as a pair `(task_id, phase_id)`.